### DISCUSSION TITLE
About volume assign request

### DISCUSSION THREAD

**Discussion by qieqieplus** [seaweedfs/seaweedfs#1996] [General]
source: weed/operation/assign_file_id.go

> 
	var requests []*VolumeAssignRequest
	requests = append(requests, primaryRequest)
	requests = append(requests, alternativeRequests...)

	var lastError error
	ret := &AssignResult{}

	for i, request := range requests {
		if request == nil {
			continue
		}

		lastError = WithMasterServerClient( ... )

		if lastError != nil {
			continue
		}

		if ret.Count <= 0 {
			lastError = fmt.Errorf("assign failure %d: %v", i+1, ret.Error)
			continue
		}

	}

	return ret, lastError

Correct me if I'm wrong, `alternativeRequests` are backups for `primaryRequest`. But it seems that even if  `primaryRequest` succeed, `alternativeRequests` will still be sent in following loops.

**Comment by chrislusf** [comment]
You are right. You can send a PR?

**Reply by qieqieplus** [reply]
Sure
